src/components/structures/RoomView.tsx

```diff
@@ -1208,12 +1211,30 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             case Action.EditEvent: {
                 // Quit early if we're trying to edit events in wrong rendering context
                 if (payload.timelineRenderingType !== this.state.timelineRenderingType) return;
+                if (payload.event && payload.event.getRoomId() !== this.state.roomId) {
+                    // If the event is in a different room (e.g. all rooms search), we need to view that room first
```

Comment:
```suggestion
                    // If the event is in a different room (e.g. because the event to be edited is being displayed in the
                    // results of an all-rooms search), we need to view that room first.
```